### ISSUE TITLE
MAX_PATTERNS=1024 cap silently produces false negatives in permitted-contents rule

### ISSUE THREAD

**Issue by saphryn** [markuplint/markuplint#3895]
## Bug Description

The `permitted-contents` rule has a hardcoded cap of 1024 patterns (`MAX_PATTERNS`) for the transparent element cross-product algorithm in `represent-transparent-nodes.ts` (line 52). When the cap is exceeded, the algorithm falls back to a conservative over-approximation that silently merges all branch children into every pattern, producing false negatives.

The issue is that this cap triggers:
- **Silently** — no warning, no debug log, no diagnostic
- **Without detection** — no metric or counter exposed to the user
- **At a non-obvious threshold** — 11+ conditional transparent siblings

## Root Cause

The cap is checked at line 216 of `packages/@markuplint/rules/src/permitted-contents/represent-transparent-nodes.ts`:

```ts
} else if (patterns.length * branchGroups.length <= MAX_PATTERNS) {
```

When `patterns.length * branchGroups.length` exceeds 1024, the `else` branch (lines 224-229) fires:

```ts
} else {
  const allChildren = branchGroups.flat();
  for (const p of patterns) {
    p.push(...allChildren);
  }
}
```

This merges ALL branch children into every pattern, losing the branch-distinction that the cross-product would preserve. The JSDoc (lines 39-48) acknowledges this:

> When the cap is exceeded, the fallback produces an **over-approximation**: all children from all branches are included in every pattern. This may cause false negatives (valid violations go undetected because the merged pattern appears to satisfy the content model).

But nothing tells the user when this fallback is active for their document.

## Impact

- **False negatives**: Valid violations go undetected when the cap triggers
- **No observability**: Users have no way to know their document triggered the cap
- **Real-world trigger**: 11+ conditional transparent siblings (e.g., `<a>` elements with `v-if`/`v-else` clones in Vue, or many conditional `<template>` nodes) hit the cap

The JSDoc says "HTML documents rarely exceed 10 conditional transparent siblings" — but framework component systems regularly produce deeply-conditional DOM structures. A component with 12+ conditional `<a>` element variants would trigger this.

## Code Location

`packages/@markuplint/rules/src/permitted-contents/represent-transparent-nodes.ts`
- Line 52: `const MAX_PATTERNS = 1024;`
- Line 216: `patterns.length * branchGroups.length <= MAX_PATTERNS`
- Lines 224-229: silent fallback (merge all branch children)
- Reference issue: #3249 (documents the original problem this was designed to solve)

## Recommended Fix

1. **Emit a diagnostic** when the cap triggers — at minimum a debug log entry
2. **Add a cap-exceeded counter** to the rule's output so CI can detect it
3. **Raise or parameterize the cap** — 1024 was conservative in the original context; it could be configurable
4. **Document the threshold** so framework users know the limit

**Comment by YusukeHirao** [comment]
Triaged and confirmed — all claims check out. The cap is the intentional trade-off introduced in the #3249 performance fix (the previous Cartesian-product algorithm took 30+ seconds on 12 transparent siblings), and the over-approximation semantics are documented in the JSDoc. But two gaps are real:

1. The fallback branch (`represent-transparent-nodes.ts`) has **no log output** — the precision loss is invisible even with debug logging enabled.
2. **No test exercises the cap-exceeded path.** Existing tests (12 non-conditional siblings) take the single-pattern path and never trigger the fallback; the fallback code is effectively untested.

**Decision:**

- Add a debug-log entry to the fallback branch so `debug` output shows when and where the over-approximation kicks in.
- Add tests that construct 11+ conditional transparent siblings to cover the fallback path, pinning both the no-false-positive guarantee and the documented false-negative behavior.
- `MAX_PATTERNS` stays at 1024 for now — making it configurable is not planned until there is a real-world report of the cap degrading results.

A PR will follow.
